docs/OBJECT-LEVEL-ENCRYPTION.md

```diff
@@ -0,0 +1,84 @@
+# OBJECT LEVEL SECURITY
+
+EVE software receives its configuration from EVC. The configuration may have confidential portions, such as passwords, acces-keys etc. Even though the configuration is exchanged over a secure channel(TLS), in case of HTTPS proxies with lawful inspection, there is a chance that these sensitive information may be exposed as plain-text for a brief period of time. In the interest of protecting these sensitive objects from getting exposed during transit, we need to deploy an additional mechanism on top of TLS to protect sensitive sections of the configuration. Using object level security, one can encrypt these objects at the origin, and by using zero-knowledge key sharing mechanism, such as ECDH, one can share the secret key used for decryption at the receiving end.  Thus a proxy, without access to this secret key, will not be able to see the actual credentials/secrets etc. 
```

Comment:
This should be broken into multiple lines for correct markdown.

Reply:
typo: "acces-keys" should be "access-keys"

Reply:
"that this sensitive information".

Reply:
I think we need to be clearer about what we are doing. It isn't _about_ individual objects, although that is how we are _implementing_ it. It is about providing TLS-style guarantees at the object body level when the envelope must be broken. I would move the last sentence or two about "Using object level security..." etc. to an implementation section.

Reply:
Yes @zed-rishabh I think we should put the motivation as EVE allowing for content inspecting proxies to be specified (reference to SECURITY.md where I'll add a note about being able to configure aCA certificate for the proxy) and the controller implementation might use load balancers with TLS termination, but there is some information in the edge-node configuration which is deemed so sensitive that it is encrypted as an object between a producing microservice in the controller and a consuming microservice in EVE, to avoid disclosing it to such trusted proxies and load balancers.